zowe/vscode-extension-for-zowe
# Migrate from yarn to pnpm

## Proposed changes



Migrate to new package manager PNPM for improved install and build times for development vs Yarn

## Release Notes




Milestone: v3.0

Changelog: Changed Yarn package manager to PNPM package manager.

## Testing this PR

For testing this PR perform the following:

- run `pnpm i --shamefully-hoist=true` (this parameter is used to maintain hositing can be set as default in npmrc) https://pnpm.io/npmrc#shamefully-hoist
- run `pnpm build`
- run `pnpm package`
- run `pnpm test`
- run `pnpm fresh-clone` (this one uses the new concurrency feature!)
- try running any npm or yarn command (this should throw an error since only pnpm commands are allowed)

Make sure to test normal ZE functionality in the dev host as well as packaged extension to make sure everything is in order as well :) 

## Metrics

`fresh-clone` command:

**Yarn**
<img width="308" alt="Screen Shot 2023-08-10 at 3 08 36 PM" src="https://github.com/zowe/vscode-extension-for-zowe/assets/68666202/f27eb213-37e5-444d-8533-17ab7b5c0191">

**PNPM**
<img width="305" alt="Screen Shot 2023-08-10 at 3 08 56 PM" src="https://github.com/zowe/vscode-extension-for-zowe/assets/68666202/6f04aef0-631d-4898-af62-352aceed0598">

`install` command:

**Yarn**
<img width="225" alt="Screen Shot 2023-08-10 at 3 11 10 PM" src="https://github.com/zowe/vscode-extension-for-zowe/assets/68666202/6f481cc5-98c5-4c58-80f8-2cd54290b93c">

**PNPM**
<img width="407" alt="Screen Shot 2023-08-10 at 3 10 17 PM" src="https://github.com/zowe/vscode-extension-for-zowe/assets/68666202/e4c7bb63-ceb8-4172-828d-f412a58831e8">

`package` command:

**Yarn**
<img width="277" alt="Screen Shot 2023-08-10 at 3 13 23 PM" src="https://github.com/zowe/vscode-extension-for-zowe/assets/68666202/c26822f5-be13-4e85-8fb0-36bfe13a33b8">

**PNPM**
<img width="274" alt="Screen Shot 2023-08-10 at 3 14 44 PM" src="https://github.com/zowe/vscode-extension-for-zowe/assets/68666202/6cc24348-5b71-49e8-a803-347406581786">

## Types of changes

What types of changes does your code introduce to Zowe Explorer?
_Put an `x` in the boxes that apply_


## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This checklist will be used as reference for both the contributor and the reviewer_


## Further comments
